src/main/java/org/mitre/inferno/Validator.java

```diff
@@ -153,6 +162,46 @@ public void loadProfile(byte[] profile) throws IOException {
     return getProfileUrls(id);
   }
 
+  private JsonObject parsePackageJson(byte[] content) throws IOException {
+    try (InputStream in = new ByteArrayInputStream(content);
+         GzipCompressorInputStream gzipIn = new GzipCompressorInputStream(in);
+         TarArchiveInputStream tarIn = new TarArchiveInputStream(gzipIn)) {
+      // Search each tar entry for package/package.json
+      TarArchiveEntry entry;
+      while ((entry = tarIn.getNextTarEntry()) != null) {
+        if (entry.isFile() && entry.getName().equals("package/package.json")) {
+          // Parse package.json
+          try (InputStreamReader reader = new InputStreamReader(tarIn);
+               BufferedReader br = new BufferedReader(reader)) {
+            return (JsonObject) JsonParser.parseReader(br);
+          }
+        }
+      }
+      throw new FileNotFoundException("Could not find package/package.json");
+    }
+  }
+
+  /**
+   * Load a Gzipped IG into the validator.
+   *
+   * @param content the Gzip-encoded contents of the IG package to be loaded
+   * @return a list of profile URLs in this IG
+   */
+  public List<String> loadPackage(byte[] content) throws Exception {
+    File temp = File.createTempFile("package", ".tgz");
+    temp.deleteOnExit();
+    try {
+      FileUtils.writeByteArrayToFile(temp, content);
+      hl7Validator.loadIg(temp.getCanonicalPath(), true);
+    } finally {
+      temp.delete();
+    }
+    // Return list of profiles for this IG
+    JsonObject packageJson = parsePackageJson(content);
+    String id = JSONUtil.str(packageJson, "name");
```

Comment:
what is `name` for?  I'm guessing this might have been deduced from other uses of this method in the hl7 library.

Reply:
Looking at the `package.json`, I assumed that the `"name"` property was just the npm package ID because it matches the ID/overall name of the package